ratpack-kafka-producer/src/main/java/smartthings/ratpack/kafka/KafkaProducerModule.java

```diff
@@ -38,6 +45,13 @@ public Properties getKafkaProperties() {
 			props.put("key.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
 			props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
 			props.put("max.block.ms", maxBlockMillis);
+			Optional.ofNullable(lingersMs).ifPresent(l -> props.put("linger.ms", l));
```

Comment:
I think I may have led you down a bad path with optional. I was thinking it would be used different which might not have made sense here since it is config.  Instead of using optional this way it might make sense to just check for null.  Such as:
```
if (lingersMs != null) {
  props.put("linger.ms", lingersMs)
}
```